rcon_adapter.py

```diff
@@ -49,3 +51,39 @@ async def addxp(self, ctx, name: str = None, skill: str = None, amount: int = No
         with Client(self.rconHost, self.rconPort, passwd=self.rconPassword, timeout=5.0) as client:
             result = client.run(f"addxp \"{name}\" {skill}={amount}")
             await ctx.send(result)
+
+    @tasks.loop(minutes=5)
+    async def update(self):
+        """sync players with syncplayers command"""
+        await self.syncplayers()
+
+    async def syncplayers(self):
+        """Syncs online players by checking number with rcon"""
+        if not self.rconPassword:
+            self.bot.log.warning('RCON password not set -- unable to syncplayers.')
+            self.update.stop()
+            return
+        self.bot.log.info('Checking rcon to see if the player counter is correct')
+        try:
+            response = await rcon(
+                'players',
+                host=self.rconHost, port=self.rconPort, passwd=self.rconPassword
+            )
+
+            # remove first line from response
+            response = ''.join(response.splitlines(keepends=True)[1:])
+
+            # update user info too
+            userHandler = self.bot.get_cog("UserHandler")
+            for user in userHandler.users.values():
```

Comment:
Could we add something here to log a warning if the current user status doesn't match what's reported from RCON? Would be good to try and work out why it's getting out of sync so this would help that

Reply:
Sounds good! Good idea

Reply:
@JonnyPtn just added some if statements for those cases. let me know if there's anything else that would make sense. Thanks!